react/features/local-recording/recording/flac/flacEncodeWorker.js

```diff
@@ -0,0 +1,416 @@
+     * @private
+     */
+    _sampleRate;
+
+    /**
+     * Bit depth (bits per sample).
+     * @private
+     */
+    _bitDepth;
+
+    /**
+     * Buffer size.
+     * @private
+     */
+    _bufferSize;
+
+    /**
+     * Buffers to store encoded bits temporarily.
+     */
+    _flacBuffers = [];
+
+    /**
+     * Length of encoded FLAC bits.
+     */
+    _flacLength = 0;
+
+    /**
+     * The current state of the {@code Encoder}.
+     */
+    _state = EncoderState.UNINTIALIZED;
+
+    /**
+     * The ready-for-grab downloadable blob.
+     */
+    _data = null;
+
+
+    /**
+     * Constructor.
+     * Note: only create instance when Flac.isReady() returns true.
+     *
+     * @param {number} sampleRate - Sample rate of the raw audio data.
+     * @param {number} bitDepth - Bit depth (bit per sample).
+     * @param {number} bufferSize - The size of each batch.
+     */
+    constructor(sampleRate, bitDepth = 16, bufferSize = 4096) {
+        if (!Flac.isReady()) {
+            throw new Error('libflac is not ready yet!');
+        }
+
+        this._sampleRate = sampleRate;
+        this._bitDepth = bitDepth;
+        this._bufferSize = bufferSize;
+
+        // create the encoder
+        this._encoderId = Flac.init_libflac_encoder(
+            this._sampleRate,
+
+            // Mono channel
+            1,
+            this._bitDepth,
+
+            FLAC_COMPRESSION_LEVEL,
+
+            // Pass 0 in becuase of unknown total samples,
+            0,
+
+            // checksum, FIXME: double-check whether this is necessary
+            true,
+
+            // Auto-determine block size (samples per frame)
+            0
+        );
+
+        if (this._encoderId === 0) {
+            throw new Error('Failed to create libflac encoder.');
+        }
+
+        // initialize the encoder
+        const initResult = Flac.init_encoder_stream(
+            this._encoderId,
+            this._onEncodedData.bind(this),
+            this._onMetadataAvailable.bind(this)
+        );
+
+        if (initResult !== 0) {
+            throw new Error('Failed to initalise libflac encoder.');
+        }
+
+        this._state = EncoderState.WORKING;
+    }
+
+    /**
+     * Receive and encode new data.
+     *
+     * @param {*} audioData - Raw audio data.
+     * @returns {void}
+     */
+    encode(audioData) {
+        if (this._state !== EncoderState.WORKING) {
+            throw new Error('Encoder is not ready or has finished.');
+        }
+
+        if (!Flac.isReady()) {
+            throw new Error('Flac not ready');
+        }
+        const bufferLength = audioData.length;
+
+        // convert to Uint32,
+        // appearantly libflac requires 32-bit signed integer input
+        // FIXME: why unsigned 32bit array?
+        const bufferI32 = new Int32Array(bufferLength);
+        const view = new DataView(bufferI32.buffer);
+        const volume = 1;
+        let index = 0;
+
+        for (let i = 0; i < bufferLength; i++) {
```

Comment:
What is audioData? An array of int16?

Reply:
Float32Array.  This has been added to the jsDoc.